stistools/add_s_region.py

```diff
@@ -0,0 +1,403 @@
+                        '52X0.5F2-R': 'L500R2',
+                        '52X2': 'L2',
+                        '52X2D1': 'L2D2',
+                        '52X2E1': 'L2E1',
+                        '52X2E2': 'L2E2',
+                        '52X2F1': 'L2F1',
+                        '52X2F1-R': 'L2R1',
+                        '52X2F2': 'L2F2',
+                        '52X2F2-R': 'L2R2',
+                        '6X0.06':'6X060',
+                        '6X0.2': '6X200',
+                        '6X0.5': '6X500',
+                        '6X6': '6X6',
+                        'BAR5': 'BAR5',
+                        'BAR10': 'BAR10',
+                        'F25CIII': '25C3',
+                        'F25CN182': '25CN182',
+                        'F25CN270': '25CN270',
+                        'F25LYA': '25LYA',
+                        'F25MGII': '25MG2',
+                        'F25ND3': '25ND3',
+                        'F25ND5': '25ND5',
+                        'F25NDQ1': '25NDQ1',
+                        'F25NDQ2': '25NDQ2',
+                        'F25NDQ3': '25NDQ3',
+                        'F25NDQ4': '25NDQ4',
+                        'F25QTZ': '25QTZ',
+                        'F25QTZD1': '25QTZD1',
+                        'F25SRF2': '25SRF2',
+                        'F25SRF2D1': '25SRF2D1',
+                        'F28X50LP': '28X50LP',
+                        'F28X50OII': '28X50O2',
+                        'F28X50OIII': '28X50O3',
+                        'WEDGEA0.6': 'WGA06',
+                        'WEDGEA1.0': 'WGA10',
+                        'WEDGEA1.8': 'WGA18',
+                        'WEDGEA2.0': 'WGA20',
+                        'WEDGEA2.5': 'WGA25',
+                        'WEDGEA2.8': 'WGA28',
+                        'WEDGEB1.0': 'WGB10',
+                        'WEDGEB1.8': 'WGB18',
+                        'WEDGEB2.0': 'WGB20',
+                        'WEDGEB2.5': 'WGB25',
+                        'WEDGEB2.8': 'WGB28'}
+
+h = pysiaf.Siaf('HST')
+
+def get_files_to_process(indir='./'):
+    """Find all _raw.fits and _tag.fits STIS files in the specified directory
+    """
+
+    file_list = []
+    file_list = file_list + glob.glob(os.path.join(indir, 'o*_raw.fits'))
+    file_list = file_list + glob.glob(os.path.join(indir, 'o*_tag.fits'))
+    return file_list
+
+def add_s_region(file, dry_run=False):
+    """Calculate the S_REGION keyword for a single STIS file. If the
+    dry_run parameter is False, set the S_REGION in the SCI extensions with
+    the calculated value.  If keyword isn't present, add it
+    """
+
+    open_mode = 'readonly'
+    if not dry_run: open_mode = 'update'
+    with fits.open(file, mode=open_mode) as f1:
+        log.info('Processing file {}'.format(file))
+        hdr0 = f1[0].header
+        detector = hdr0['DETECTOR']
+        aperture = hdr0['APERTURE']
+        propaper = hdr0['PROPAPER']
+        for ext in f1[1:]:
+            if ext.header['EXTNAME'] == 'SCI':
```

Comment:
`tag` files have extname=`EVENTS` instead of `SCI`.

Reply:
Thanks!  There's a bunch of other stuff related to timetag data that I needed to fix, e.g. WCS keyword names